outputs.tf

```diff
@@ -7,9 +7,12 @@ output "key_protect_guid" {
   value       = local.key_protect_guid
 }
 
+
+##############################################################################
+
 output "key_protect_name" {
-  description = "Name of the Key Protect instance"
-  value       = local.key_protect_name
+  description = "Key Protect Name"
+  value       = var.key_protect_instance_name
```

Comment:
as per call, use the resource output here, not input variable

Reply:
After some testing, I realized that we cannot use the resource output for this as it is behind a count, so if we try to have the output be `module.key_protect[0].key_protect_name` it will give an error if the `create_key_protect_instance` flag is false. As this is the case the output is currently still set as `var.key_protect_instance_name`

Reply:
We should be able to handle this with logic. Default to null if `create_key_protect_instance` is set to `false`. See example:
https://github.com/terraform-ibm-modules/terraform-ibm-observability-instances/blob/main/outputs.tf#L20